### DISCUSSION TITLE
Automatically checking if cutadapt_trim_3p is needed

### DISCUSSION THREAD

**Discussion by marchoeppner** [bio-raum/FooDMe2#25] [General]
@gregdenay I gave this idea a bit of a go.  We could absolutely make it work for paired-end Illumina data - by parsing the JSON report that comes ouf of FastP (`insert size` versus `mean read length`). 

But I don't see a way to implement this for any type of single-end read data, including ONT. So my feeling is that we should probably not do it and leave it to the user. Not a fan of different types of data requiring different types of user intervention at least. 

Thoughts?

**Comment by marchoeppner** [comment]
As a bit of a compromise, I have added a check for paired-end data that will print a warning to the screen, by sample, if the insert size is greater than the mean read length minus 20bp (random number, that should somewhat account for the PCR primers that have yet to be removed). 

```NEXTFLOW
if (!params.cutadapt_trim_3p) {
        ch_json_with_insert_size.filter {m,j -> m.insert_size > (m.mean_read_length - 20)}.subscribe { m,j ->
            log.warn "${m.sample_id} - the mean insert size seems to be close to or greater than the mean read length. Should you perhaps use --cutadapt_trim_3p?"
        }
    }
```

So that we at least make some use of the information. Or would that be confusing?

**Comment by gregdenay** [comment]
I think this is fine. THe warning is there for people to see and it should make them think about it. Single end users should be mainly ONT (is anyone still doing illumina SE?) and they should know to trim on both ends for amplicon sequencing.

If users forget the parameter the results are going to be crap anyways so they would have to read the documentation :)
We should add a mention of this in a FAQ/Troobleshooting.

Another solution would be to check how many reads would be processed by a second pass trimming on the 3' end, and decide based on a %cutoff, but that is far from transparent and I can't say I like it.

**Reply by marchoeppner** [reply]
Done. You can test it by running the FooDMe test data with `--primers_fa` , `--db lrna` but without `--cutadapt_trim_3p`

**Reply by marchoeppner** [reply]
With that said, should we also have a warning for any type of single-end data when `--cutadapt_trim_3p` is not specified? You are obviously right that this will almost always mean that both primers are included in a read.

**Reply by gregdenay** [reply]
That is a good idea!

**Reply by marchoeppner** [reply]
Also done. Now we only need to find single-end data for testing ;)
